### PR TITLE
Avoid abbreviation in Python API

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**

### CODE REVIEW

File: src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi [hunk 1]
```diff
@@ -338,8 +338,9 @@ def server_certificate_config_ssl(pem_root_certs, pem_key_cert_pairs):
 def server_credentials_ssl_dynamic_cert_config(initial_cert_config,
                                                cert_config_fetcher,
                                                bint force_client_auth):
-  if not isinstance(initial_cert_config, grpc.ServerCertificateConfig):
-    raise TypeError('initial_cert_config must be a grpc.ServerCertificateConfig')
+  if not isinstance(initial_cert_config, grpc.ServerCertificateConfiguration):
+    raise TypeError(
+        'initial_cert_config must be a grpc.ServerCertificateConfiguration')
```
Comment (cauthu) [inline src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi]: nit: we say `initial_cert_config` here where as the user is passing `initial_certificate_configuration` to while calling `dynamic_ssl_server_credentials()`. i guess it's close enough that they can figure it out? or, should we move this type check one layer up, into `__init__.py`?

Comment (nathanielmanistaatgoogle) [reply src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi]: Thanks for noticing this.

In a separate change we should move this type check one layer up.

**Comment by mehrdada** [comment]
I suggest find-and-replacing all occurrences of SSL with TLS, at least in comments, if the already-existing API names understandably cannot get fixed.

File: src/python/grpcio/grpc/__init__.py [hunk 2]
```diff
@@ -1277,46 +1279,47 @@ def ssl_server_certificate_config(private_key_certificate_chain_pairs,
...
     Args:
-      initial_cert_config (ServerCertificateConfig): the certificate
-        config with which the server will be initialized.
-      cert_config_fetcher (callable): a callable that takes no
-        arguments and should return a ServerCertificateConfig to
+      initial_certificate_configuration (ServerCertificateConfiguration): The
+        certificate configuration with which the server will be initialized.
+      certificate_configuration_fetcher (callable): A callable that takes no
+        arguments and should return a ServerCertificateConfiguration to
         replace the server's current cert, or None for no change
```
[the remark this answers is not in the document]
Comment (nathanielmanistaatgoogle) [reply src/python/grpcio/grpc/__init__.py]: Fixed; thanks for the catch.

**Comment by nathanielmanistaatgoogle** [comment]
Cool beans; "SSL" → "TLS" renaming to be discussed in [issue 13481](https://github.com/grpc/grpc/issues/13481).

**Comment by nathanielmanistaatgoogle** [comment]
Pre-merge test assessment:
* Basic Tests (MacOS, Windows) (dbg, opt): [issue 13426](https://github.com/grpc/grpc/issues/13426).
* Interop Cloud-To-Cloud Tests: [issue 13482](https://github.com/grpc/grpc/issues/13482).

**Pull request merged**